### DISCUSSION TITLE
Reporting as general User & confusing total's

### DISCUSSION THREAD

**Discussion by xstable** [kimai/kimai#4162] [Questions & Answers]
If you are logged in as general user (who general only see his own stuff), he see the working-hours for the whole project in the Reporting-Area. 

![image](https://github.com/kimai/kimai/assets/1778068/769c3ae1-ab5c-47a4-bc80-a6aa1a79242d)

And here a Snippet of the Project-Overview as Admin to see, that this number is the Project-Total.: 
![image](https://github.com/kimai/kimai/assets/1778068/169c7b52-735d-4c3c-a156-0ce09872f3ff)

For "per Week", "today" and other filters, it works as expected (as - in this case)- the user not yet worked today or this week on this project: 
![image](https://github.com/kimai/kimai/assets/1778068/d82b7d46-dd2c-4b17-b7c1-bcb0adae66d1)

That is very confusing for the user, as some of the values are "his own" ones, and other the total of the project. 

I suggest to make sure, that "non-teamlead" users only see reports about their own stuff, and no "project-total".

**Comment by kevinpapst** [comment]
I don't see a problem. 
Regular users do not have the permission to see this report.
They have extended permissions (e.g. project_time, project_time_team) granted by you.

The admin can open up this data, that is what happened here.

**Reply by xstable** [reply]
I'm logged in as this user (Role: User): 
![image](https://github.com/kimai/kimai/assets/1778068/2fe56fe3-b2b2-4694-b0d8-2c71c6598a9f)

With Assignment to this Team: 
![image](https://github.com/kimai/kimai/assets/1778068/2242c18b-99e0-4b00-b685-fddd2ec68f65)

I didn't change anything in the role-set of the user, admin or teamlead role, but the screenshot in my initial post from this "User-Role" account.

**Reply by kevinpapst** [reply]
I doubt that this has changed between 1.x and 2.0, but I am not giving support for this outdated Kimai version anyway.
Upgrade to the latest major version and we can talk again.

**Reply by xstable** [reply]
Here is a video that shows the issue: https://somup.com/c0ieq7z2UA

Oh, didn't noticed, that my version is outdated (O.o) That might explain some of the issues. 
I hope upgrading is not too hard. Will give it a try soon.

Thanks for your helping hands.

**Reply by kevinpapst** [reply]
The code clearly says that this report is only available to users with certain permissions (budget_any):
https://github.com/kimai/kimai/blob/1.x/src/Reporting/ReportingService.php#L57

This maps to another permission (here likely project_time or project_time_team)
https://github.com/kimai/kimai/blob/1.x/src/Voter/EntityMultiRoleVoter.php#L108

Which is only contained in the permission set PROJECTS:
https://github.com/kimai/kimai/blob/1.x/config/packages/kimai.yaml#L84C13-L84C21

Which itself is only listed as `@PROJECTS` in the sets for ROLE_ADMIN and ROLE_SUPER_ADMIN.
https://github.com/kimai/kimai/blob/1.x/config/packages/kimai.yaml#L116

There is only one explanation: you changed the role permissions for the default user role (or your user is in another non-default role), there is no other reason why this report would be available in the first place.
